app/src/androidTest/java/com/github/ybecker/epforuml/MapFragmentTest.kt

```diff
@@ -87,4 +91,111 @@ class MapFragmentTest {
             .perform(click())
         openContextualActionModeOverflowMenu()
     }
+
+
+    @Test
+    fun checkUserCoordinates() {
+
+        // Sign in user
+        scenario.onActivity { MockAuthenticator(it).signIn().join() }
+
+        val user = DatabaseManager.user
+
+        var lat = user?.latitude
+        var lon = user?.longitude
+
+        if (lat != null) {
+            TestCase.assertEquals(initialLat, lat.roundToInt())
+        }
+        if (lon != null) {
+            TestCase.assertEquals(initialLon, lon.roundToInt())
+        }
+
+        // Open navigation and click on map
+        Espresso.onView(ViewMatchers.withContentDescription(R.string.open))
+            .perform(click())
+        Espresso.onView(ViewMatchers.withId(R.id.nav_map))
+            .perform(click())
+
+
+        // Open overflow menu and enable location sharing
+        openContextualActionModeOverflowMenu()
+        Espresso.onView(ViewMatchers.withText(R.string.share_position))
+            .check(ViewAssertions.matches(not(ViewMatchers.isChecked())))
+            .perform(click())
+
+
+        Thread.sleep(2000)
```

Comment:
Instead of user Thread.sleep you could have use the onViewWithTimout in app/Unit !